src/main/java/net/dries007/tfc/objects/te/TEFirePit.java

```diff
@@ -322,7 +327,57 @@ private void handleInputMelting(ItemStack stack)
+                            if (output1Cap != null)
+                            {
+                                long earliest = Math.max(output1Cap.getCreationDate(), foodCap.getCreationDate());
+                                output1Cap.setCreationDate(earliest);
+                                int merge = Math.min(slot1Stack.getMaxStackSize(), outputStack.getCount() + slot1Stack.getCount());
+                                // Why not inventory#insertItem you ask?
+                                // 1 - we would need to update the creation date of the outputStack first before inserting, which would undesirable propagate to the second slot or even further when spit out to the world
+                                // 2 - if one of then have a trait while the other don't (ie: salted), they will not merge even after updating the creation date
+                                // 3 - It is possible to fix all of the above, but you will need to handle all possible scenarios which would make you want to just spit out the thing to the world.
+                                outputStack.shrink(merge - slot1Stack.getCount());
+                                slot1Stack.setCount(merge);
+                            }
+                        }
+
+                        //Second slot merging (remaining) if possible
+                        ItemStack slot2Stack = inventory.getStackInSlot(SLOT_OUTPUT_2);
+                        if (!outputStack.isEmpty() && outputStack.isItemEqual(slot2Stack) && slot2Stack.getCount() < slot2Stack.getMaxStackSize())
+                        {
+                            // If both are the same food and are mergeable, ensure both have the same earliest creation date
+                            IFood output2Cap = slot2Stack.getCapability(CapabilityFood.CAPABILITY, null);
+                            if (output2Cap != null)
+                            {
+                                long earliest = Math.max(output2Cap.getCreationDate(), foodCap.getCreationDate());
+                                output2Cap.setCreationDate(earliest);
+                                int merge = Math.min(slot2Stack.getMaxStackSize(), outputStack.getCount() + slot2Stack.getCount());
+                                outputStack.shrink(merge - slot2Stack.getCount());
+                                slot2Stack.setCount(merge);
+                            }
+                        }
+                    }
+                    if (!outputStack.isEmpty())
+                    {
+                        // If we got here, we failed merging foods, or this wasn't a food at all
+                        // Spit out the item in a random position. This is gonna avoid the food catching fire in the fire pit.
+                        // Unless this fire pit is placed near walls.
+                        Helpers.spawnItemStack(world, pos.offset(EnumFacing.byHorizontalIndex(Constants.RNG.nextInt(4))), outputStack);
```

Comment:
Instead of throwing it in the world, just replace the result in the input slot. The result at the moment is essentially the same as current, just with a random "did you build your fireplace an open location" chance. Not an intuitive or friendly mechanic.

Reply:
Came across a design problem here: While it is possible to just output to the input slot when the recipe is 1 item input -> 1 item output (and both have the same max stack), if, for example, we have 1 item output -> 2 item outputs, how can I put it into the input slot? The way it is implemented currently `handleInputMelting` works by converting the `cachedRecipe` input->output one at a time.

Another suggestion for this i could think of is working exactly like barrels, storing a queue of items if both slots are full.